modules/broker-core/src/main/java/io/ballerina/messaging/broker/core/events/BrokerEventManager.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+
+package io.ballerina.messaging.broker.core.events;
+
+import io.ballerina.messaging.broker.core.QueueHandler;
+
+/**
+ * BrokerEventManager handles all the events related to Broker class.
+ */
+public interface BrokerEventManager {
```

Comment:
Why do we need an event manager object? Instead can't we pass the event publisher to the places where the events originate. If we do it like this won't event manager object be a central point with a lot of methods?